### PR TITLE
Fix #85 Add Disable Purge Option

### LINKED ISSUE #85: [Feature Request]: Add ability to optionally shutoff available space check
### Pacifica Data Cart version
0.4.0
See code hovering under https://github.com/pacifica/pacifica-cartd/blob/master/pacifica/cartd/utils.py#L80

### Scenario:
Trying to use a network mounted share to do cart generation. Due to limitation of certain protocols, I am unable to determine the true size of the attached share. Due to this limitation and being forced to calculate if there is enough space, I am unable to generate a cart and house it on these kinds of mounts.

### Steps to Reproduce:
Utilize a cifs based mount for cart generation.

### Expected Result:
Cart should generate regardless of available storage if the right flag OR env var is specified.

### Actual Result:
Cart generation fails silently with Not enough space available to generate cart because the methods used are unable to determine actual available size of underlying storage (i.e. 0 bytes available but it isn't true).

### PR DESCRIPTION
**Pull request by dmlb2000**
### Description

This adds a feature to disable the auto purge feature of the cart
backend tasks.

Signed-off-by: David Brown <dmlb2000@gmail.com>

### Issues Resolved

Fix #85 
### Check List

- [ ] All tests pass. See <https://github.com/pacifica/pacifica-docs/blob/master/development/TESTING.md>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by GitBytes** [comment]
@dmlb2000 So does this allow me to disable the `check_space_requirements` method. It doesn't appear to?

File: pacifica/cartd/utils.py [hunk 1]
```diff
@@ -106,6 +107,8 @@ def check_space_requirements(self, cart_file, mycart, size_needed, deleted_flag)
         space. It will delete a cart first, then call  itself
         until either there is enough space or there is no carts to delete
         """
+        if not self._lru_purge:
```
Comment (dmlb2000) [inline pacifica/cartd/utils.py:110]: @GitBytes this check short circuits the function you want.

Comment (GitBytes) [reply pacifica/cartd/utils.py:110]: @dmlb2000 Ah ok, I didn't read it quite right then. Cool! Thanks!

**Review by GitBytes — APPROVED** [review]
Reviewed and LGTM once tests are good to go. Cheers!

**Review by GitBytes — APPROVED** [review]

**Comment by GitBytes** [comment]
@dmlb2000 I guess my review doesn't count to clear the PR. Just FYI.

**Pull request merged**